docs/rancher/02-setup-k3s/README.md

```diff
@@ -0,0 +1,144 @@
+
+- Installed k3s on your Ubuntu server, ready for Rancher.
+- Configured your local environment for seamless k3s cluster management.
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Step 1 - Installing and Verifying k3s](#step-1---installing-and-verifying-k3s)
+- [Step 2 - Transferring the kubeconfig File](#step-2---transferring-the-kubeconfig-file)
+- [Step 3 - Setting Up DNS for the Server](#step-3---setting-up-dns-for-the-server)
+- [Step 4 - Customizing the kubeconfig File](#step-4---customizing-the-kubeconfig-file)
+- [Step 5 - Merging Configuration Files for Cluster Management](#step-5---merging-configuration-files-for-cluster-management)
+- [Conclusion](#conclusion)
+
+## Prerequisites
+
+Before starting, ensure you have:
+
+1. An Ubuntu server set up and secured. For instructions, refer to "[How to Set Up and Secure Your Ubuntu Server](../01-setup-ubuntu/README.md)"
+2. Terminal or SSH client access to your local machine.
+
+## Step 1 - Installing and Verifying k3s
+
+Before installing k3s, ensure you are connected to your Ubuntu server.
+
+```bash
+ssh root@<ip_address>
+```
+
+Replace `<ip_address>` with the actual IP address of your server. Once connected, you can begin the installation of k3s. The version `v1.24.14+k3s1` is specifically chosen to ensure compatibility with Rancher, as recommended in the [Rancher Quickstart Guide](https://github.com/rancher/quickstart/tree/master/rancher/rancher-common).
+
+```bash
+curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.24.14+k3s1 sh -s - server --cluster-init
+```
+
+After the installation completes, verify that your cluster is operational:
+
+```bash
+k3s kubectl get nodes
+k3s kubectl get pods --all-namespaces
+```
+
+You should see nodes marked as `Ready`, and pods should be in `Running` or `Completed` states, indicating that k3s is installed and functioning correctly on your server.
+
+## Step 2 - Transferring the kubeconfig File
+
+Transfer the kubeconfig file to manage the cluster remotely:
+
+```bash
+scp root@<ip_address>:/etc/rancher/k3s/k3s.yaml ~/.kube/k3s-murm-rancher.yaml
+```
+
+Replace `<ip_address>` with your server's actual IP. This step is crucial for remote cluster administration.
+
+## Step 3 - Setting Up DNS for the Server
+
+Before customizing your kubeconfig file, ensure your server is accessible via a DNS name. This involves adding an A record in your DNS management interface pointing to your server’s IP address.
+
+- Log in to your DNS provider's management console.
+- Navigate to the DNS settings section for your domain.
+- Add an A record with the following details:
+  - **Host:** The subdomain or prefix you wish to use (e.g., `k3s` or `rancher`).
+  - **Points to:** The IP address of your Ubuntu server.
+  - **TTL:** Set according to your preference or provider's default.
+
+This DNS setup facilitates access to your server using a memorable URL instead of an IP address.
```

Comment:
Most importantly, if DNS is not used, the backup and restore processes will not work if the IP address changes.